### DISCUSSION TITLE
[Feature] Make Kafka consumer's traceId same as provider's traceId

### DISCUSSION THREAD

**Discussion by chenzz** [apache/skywalking#8571] [Q&A] [answered]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Currently, 
kafka consumer generate a new traceId, which differs from provider traceId, and is inconvenient when trace problem.

Hope,
Make Kafka consumer's traceId same as provider's traceId

### Use case

Make Kafka consumer's traceId same as provider's traceId

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment, marked as answer]
I think your understanding is not accurate. One different trace is generated just temporarily, because there is no chance to know a consuming happens. Once consuming happens, it overrides the generated trace id. 
This is the reality, not about inconvenient or not.

You could think deeply about how Kafka consumer works.

**Reply by chenzz** [reply]
> Once consuming happens, it overrides the generated trace id.

I did not see the logic you describe above, did I miss that?

See `org.apache.skywalking.apm.plugin.kafka.KafkaConsumerInterceptor#afterMethod`
<img width="1206" alt="image" src="https://user-images.githubusercontent.com/5676311/155108463-5ed51ccc-a544-4d51-81c5-9f2f2bfdd7dd.png">

**Reply by wu-sheng** [reply]
I didn't write or am not familiar with Kafka interceptor, but I think it is this.

https://github.com/apache/skywalking-java/blob/f0c22f32dd1493c03b01013ddac93e861902d153/apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaConsumerInterceptor.java#L95

**Reply by lujiajing1126** [reply]
The reason why it is impossible to have the same traceID is essentially how the streaming system works.

As you may find in the code given by Wu-Sheng above, `traceID` (stored in the header) are probably diverse due to the batch polling.

**Reply by superido** [reply]
你好，我遇到一个问题，快速调用一个spring action多次（产生多个traceid A，B，C...）但是消费者那边的traceID都是A。如果我一个一个降低访问速度的话traceid就没有问题。
I encountered an issue. there are several requests come in, generated multiple traceids (A,B,C), but the traceID of consumers are only A. but it is right when I have interval access to it.

SW：V9.1
MQ：Kafka V2.7
SpringCloud structure

> The reason why it is impossible to have the same traceID is essentially how the streaming system works.
> 
> As you may find in the code given by Wu-Sheng above, `traceID` (stored in the header) are probably diverse due to the batch polling.
